api/middleware.go

```diff
@@ -96,3 +98,33 @@ func stripServerSideQueryParams(r *http.Request) {
 	q.Del("refstr")
 	r.URL.RawQuery = q.Encode()
 }
+
+const (
+	bearerPrefix        = "Bearer "
+	authorizationHeader = "authorization"
+)
+
+// OAuthTokenMiddleware parses any "authorization" header containing a Bearer
+// token & adds it to the request context
+func OAuthTokenMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		reqToken := r.Header.Get(authorizationHeader)
```

Comment:
Can we add a TODO to extract this functionality and re-use where needed, by now we have several code bases that are doing 99% the same as here.